Error when trying to run batch

When trying to run batch mode the action fails because it's also passing a repo list 

```bash
[dotenv@17.3.1] injecting env (0) from .env -- tip: 🛡️ auth for agents: https://vestauth.com/
Version: 3.1.0
Validating options...
file:///home/runner/.local/share/gh/extensions/gh-repo-stats-plus/dist/index.js:7011
            throw new Error('Batch mode (--batch-size) cannot be used with --repo-list. Batch mode generates its own repo list.');
                  ^

Error: Batch mode (--batch-size) cannot be used with --repo-list. Batch mode generates its own repo list.
    at validate$4 (file:///home/runner/.local/share/gh/extensions/gh-repo-stats-plus/dist/index.js:7011:19)
    at Command.<anonymous> (file:///home/runner/.local/share/gh/extensions/gh-repo-stats-plus/dist/index.js:7110:5)
    at Command.listener [as _actionHandler] (/home/runner/.local/share/gh/extensions/gh-repo-stats-plus/node_modules/commander/lib/command.js:568:17)
    at /home/runner/.local/share/gh/extensions/gh-repo-stats-plus/node_modules/commander/lib/command.js:1604:14
    at Command._chainOrCall (/home/runner/.local/share/gh/extensions/gh-repo-stats-plus/node_modules/commander/lib/command.js:1488:12)
    at Command._parseCommand (/home/runner/.local/share/gh/extensions/gh-repo-stats-plus/node_modules/commander/lib/command.js:1603:27)
    at /home/runner/.local/share/gh/extensions/gh-repo-stats-plus/node_modules/commander/lib/command.js:1367:27
    at Command._chainOrCall (/home/runner/.local/share/gh/extensions/gh-repo-stats-plus/node_modules/commander/lib/command.js:1488:12)
    at Command._dispatchSubcommand (/home/runner/.local/share/gh/extensions/gh-repo-stats-plus/node_modules/commander/lib/command.js:1363:25)
    at Command._parseCommand (/home/runner/.local/share/gh/extensions/gh-repo-stats-plus/node_modules/commander/lib/command.js:1559:19)

Node.js v25.8.2
Error: Process completed with exit code 1.
```

--- a/README.md
+++ b/README.md
@@ -20,25 +20,25 @@ In both cases, the `github-token` input (typically `${{ secrets.GITHUB_TOKEN }}`
 
 ## Inputs
 
-| Input                    | Description                                                                                                               | Required | Default                  |
-| ------------------------ | ------------------------------------------------------------------------------------------------------------------------- | -------- | ------------------------ |
-| `type`                   | Type of stats gathering: `repository`, `organization`, `project-stats`, `app-install-stats`, or `combine`                 | No       | `repository`             |
-| `github-token`           | GitHub token for authentication (e.g., `github.token`)                                                                    | **Yes**  |                          |
-| `ghec-token`             | GitHub Enterprise Cloud token (used to download dependencies from GHEC if not on github.com)                              | No       | `""`                     |
-| `access-token`           | Personal access token with repo access for gathering stats                                                                | No       | `""`                     |
-| `github-app-id`          | GitHub App ID for authentication (requires `github-app-private-key`)                                                      | No       | `""`                     |
-| `github-app-private-key` | GitHub App private key for authentication (requires `github-app-id`)                                                      | No       | `""`                     |
-| `organization`           | Organization or owner name                                                                                                | **Yes**  |                          |
-| `repository`             | Repository name (required if `type` is `repository`)                                                                      | No       | `""`                     |
-| `output-dir`             | Directory where output files will be stored                                                                               | No       | `output`                 |
-| `run-migration-audit`    | Whether to run migration audit (`true`/`false`)                                                                           | No       | `false`                  |
-| `node-version`           | Node.js version to use                                                                                                    | No       | `25`                     |
-| `base-url`               | GitHub API base URL                                                                                                       | No       | `https://api.github.com` |
-| `skip-tls-verification`  | Skip TLS certificate verification for the target GitHub instance (use for GHES with self-signed certs or IP-based access) | No       | `false`                  |
-| `retention-days`         | Number of days to retain uploaded artifacts                                                                               | No       | `7`                      |
+| Input                    | Description                                                                                                                                                             | Required | Default                  |
+| ------------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------- | -------- | ------------------------ |
+| `type`                   | Type of stats gathering: `repository`, `organization`, `project-stats`, `app-install-stats`, or `combine`                                                               | No       | `repository`             |
+| `github-token`           | GitHub token for authentication (e.g., `github.token`)                                                                                                                  | **Yes**  |                          |
+| `ghec-token`             | GitHub Enterprise Cloud token (used to download dependencies from GHEC if not on github.com)                                                                            | No       | `""`                     |
+| `access-token`           | Personal access token with repo access for gathering stats                                                                                                              | No       | `""`                     |
+| `github-app-id`          | GitHub App ID for authentication (requires `github-app-private-key`)                                                                                                    | No       | `""`                     |
+| `github-app-private-key` | GitHub App private key for authentication (requires `github-app-id`)                                                                                                    | No       | `""`                     |
+| `organization`           | Organization or owner name                                                                                                                                              | **Yes**  |                          |
+| `repository`             | Repository name (required if `type` is `repository`)                                                                                                                    | No       | `""`                     |
+| `output-dir`             | Directory where output files will be stored                                                                                                                             | No       | `output`                 |
+| `run-migration-audit`    | Whether to run migration audit (`true`/`false`)                                                                                                                         | No       | `false`                  |
+| `node-version`           | Node.js version to use                                                                                                                                                  | No       | `25`                     |
+| `base-url`               | GitHub API base URL                                                                                                                                                     | No       | `https://api.github.com` |
+| `skip-tls-verification`  | Skip TLS certificate verification for the target GitHub instance (use for GHES with self-signed certs or IP-based access)                                               | No       | `false`                  |
+| `retention-days`         | Number of days to retain uploaded artifacts                                                                                                                             | No       | `7`                      |
 | `batch-size`             | Number of repositories per batch (enables batch processing for large organizations). **Cannot be combined with `repository`** — batch mode generates its own repo list. | No       | `""`                     |
-| `batch-index`            | Zero-based batch index (used with `batch-size` for parallel matrix jobs)                                                  | No       | `""`                     |
-| `batch-delay`            | Delay in seconds multiplied by batch index to stagger API requests and avoid rate limits                                  | No       | `""`                     |
+| `batch-index`            | Zero-based batch index (used with `batch-size` for parallel matrix jobs)                                                                                                | No       | `""`                     |
+| `batch-delay`            | Delay in seconds multiplied by batch index to stagger API requests and avoid rate limits                                                                                | No       | `""`                     |
 
 ## Outputs
 

--- a/README.md
+++ b/README.md
@@ -36,7 +36,7 @@ In both cases, the `github-token` input (typically `${{ secrets.GITHUB_TOKEN }}`
 | `base-url`               | GitHub API base URL                                                                                                       | No       | `https://api.github.com` |
 | `skip-tls-verification`  | Skip TLS certificate verification for the target GitHub instance (use for GHES with self-signed certs or IP-based access) | No       | `false`                  |
 | `retention-days`         | Number of days to retain uploaded artifacts                                                                               | No       | `7`                      |
-| `batch-size`             | Number of repositories per batch (enables batch processing for large organizations)                                       | No       | `""`                     |
+| `batch-size`             | Number of repositories per batch (enables batch processing for large organizations). **Cannot be combined with `repository`** — batch mode generates its own repo list. | No       | `""`                     |
 | `batch-index`            | Zero-based batch index (used with `batch-size` for parallel matrix jobs)                                                  | No       | `""`                     |
 | `batch-delay`            | Delay in seconds multiplied by batch index to stagger API requests and avoid rate limits                                  | No       | `""`                     |
 

--- a/action.yml
+++ b/action.yml
@@ -383,7 +383,8 @@ runs:
         if [[ -n "${REPO_LIST_FILE}" ]] && [[ -z "${BATCH_SIZE}" ]]; then
           REPO_LIST_ARGS+=(--repo-list "${REPO_LIST_FILE}")
         elif [[ -n "${REPO_LIST_FILE}" ]] && [[ -n "${BATCH_SIZE}" ]]; then
-          echo "::warning::--repo-list is ignored when batch mode (--batch-size) is enabled. Batch mode generates its own repo list."
+          echo "::error::Batch mode (--batch-size) cannot be combined with a repository list (from the 'repository' input). Remove one of them."
+          exit 1
         fi
 
         gh repo-stats-plus repo-stats \
@@ -432,7 +433,8 @@ runs:
         if [[ -n "${REPO_LIST_FILE}" ]] && [[ -z "${BATCH_SIZE}" ]]; then
           REPO_LIST_ARGS+=(--repo-list "${REPO_LIST_FILE}")
         elif [[ -n "${REPO_LIST_FILE}" ]] && [[ -n "${BATCH_SIZE}" ]]; then
-          echo "::warning::--repo-list is ignored when batch mode (--batch-size) is enabled. Batch mode generates its own repo list."
+          echo "::error::Batch mode (--batch-size) cannot be combined with a repository list (from the 'repository' input). Remove one of them."
+          exit 1
         fi
 
         gh repo-stats-plus project-stats \

--- a/action.yml
+++ b/action.yml
@@ -378,6 +378,14 @@ runs:
           unset BATCH_DELAY
         fi
 
+        REPO_LIST_ARGS=()
+        REPO_LIST_FILE="${{ steps.setup-paths.outputs.file }}"
+        if [[ -n "${REPO_LIST_FILE}" ]] && [[ -z "${BATCH_SIZE}" ]]; then
+          REPO_LIST_ARGS+=(--repo-list "${REPO_LIST_FILE}")
+        elif [[ -n "${REPO_LIST_FILE}" ]] && [[ -n "${BATCH_SIZE}" ]]; then
+          echo "::warning::--repo-list is ignored when batch mode (--batch-size) is enabled. Batch mode generates its own repo list."
+        fi
+
         gh repo-stats-plus repo-stats \
           --org-name "${{ inputs.organization }}" \
           --access-token "${{ inputs.access-token }}" \
@@ -385,8 +393,8 @@ runs:
           --private-key "${{ inputs.github-app-private-key }}" \
           --app-installation-id "${{ steps.app-install.outputs.installation-id }}" \
           --base-url "${{ inputs.base-url }}" \
-          --repo-list "${{ steps.setup-paths.outputs.file }}" \
           --output-dir "${{ steps.setup-paths.outputs.output_dir }}" \
+          "${REPO_LIST_ARGS[@]}" \
           "${BATCH_ARGS[@]}"
 
         echo "✓ Repository statistics gathered successfully"
@@ -419,6 +427,14 @@ runs:
           unset BATCH_DELAY
         fi
 
+        REPO_LIST_ARGS=()
+        REPO_LIST_FILE="${{ steps.setup-paths.outputs.file }}"
+        if [[ -n "${REPO_LIST_FILE}" ]] && [[ -z "${BATCH_SIZE}" ]]; then
+          REPO_LIST_ARGS+=(--repo-list "${REPO_LIST_FILE}")
+        elif [[ -n "${REPO_LIST_FILE}" ]] && [[ -n "${BATCH_SIZE}" ]]; then
+          echo "::warning::--repo-list is ignored when batch mode (--batch-size) is enabled. Batch mode generates its own repo list."
+        fi
+
         gh repo-stats-plus project-stats \
           --org-name "${{ inputs.organization }}" \
           --access-token "${{ inputs.access-token }}" \
@@ -426,8 +442,8 @@ runs:
           --private-key "${{ inputs.github-app-private-key }}" \
           --app-installation-id "${{ steps.app-install.outputs.installation-id }}" \
           --base-url "${{ inputs.base-url }}" \
-          --repo-list "${{ steps.setup-paths.outputs.file }}" \
           --output-dir "${{ steps.setup-paths.outputs.output_dir }}" \
+          "${REPO_LIST_ARGS[@]}" \
           "${BATCH_ARGS[@]}"
 
         echo "✓ Project statistics gathered successfully"

